contracts/StakeLockContract.sol

```diff
@@ -28,74 +29,78 @@ contract StakeLockContract {
      */
     mapping(address => lockedToken) private locked;
     mapping(address => uint) private indices;
-    address[] private accountList;
+    address[] private addressList;
 
     event Locked(address indexed _of, uint _amount, uint256 _epoch);
     event Unlocked(address indexed account, uint index);
 
-    /**
-     * @dev Returns total tokens held by an address (locked + transferable)
-     * @param _of The address to query the total balance of
-     */
-    function lockedBalanceOf(address _of)
-        public
-        view
-        returns (uint256 amount)
-    {
-        amount = locked[_of]._amount;
-    }
-
     /**
      * @dev Locks a specified amount of tokens against an address
      *      starting at the specific epoch
-     * @param _epoch The epoch where the lock happens
      */
-    function lock(uint256 _epoch)
+    function lock()
         public
         payable
         returns (bool)
     {
         // If tokens are already locked, then functions extendLock or
         // increaseLockAmount should be used to make any changes
-        require(lockedBalanceOf(msg.sender) == 0, ALREADY_LOCKED);
+        require(balanceOf(msg.sender) == 0, ALREADY_LOCKED);
         require(msg.value != 0, AMOUNT_ZERO);
 
-        locked[msg.sender] = lockedToken(msg.value, block.number, currentEpoch(), 1);
+        // By default, the tokens can only be locked for one locking period.
+        locked[msg.sender] = lockedToken(msg.value, block.number, currentEpoch(), 1, addressList.push(msg.sender) - 1);
 
-        emit Locked(msg.sender, msg.value, _epoch);
+        emit Locked(msg.sender, msg.value, currentEpoch());
         return true;
     }
 
     /**
      * @dev Unlocks the unlockable tokens of a specified address
-     * @param _requestor Address of user, claiming back unlockable tokens
      */
-    function unlock(address payable _requestor)
+    function unlock()
         public
         returns (uint256 unlockableTokens)
     {
-        unlockableTokens = getUnlockableTokens(_requestor);
+        unlockableTokens = getUnlockableTokens(msg.sender);  // For now the unlockableTokens is all the tokens for a address
+        require(unlockableTokens != 0, NO_TOKEN_UNLOCKABLE);
 
-        if (unlockableTokens != 0) {
-            _requestor.transfer(unlockableTokens);
-        }
+        uint indexToRemove = locked[msg.sender]._index;
+        delete locked[msg.sender];
+        addressList[indexToRemove] = addressList[addressList.length - 1];
+        locked[addressList[addressList.length - 1]]._index = indexToRemove;
```

Comment:
why do you need two structure, locked, and addressList to save the data?  why don't just put the address in the locked list?

Reply:
Solidity's mapping only store values keyed by the hash of the keys. The original keys are not stored so there is no way to enumerate a map. Have to add extra array to store the keys: https://ethereum.stackexchange.com/questions/15337/can-we-get-all-elements-stored-in-a-mapping-in-the-contract